File: XPile 3-17/Programs/test_dynamic.js

function main() {
    let nums = [];
    nums.push(10);
    nums.push(20);
    nums.push(30);
    console.log(nums.length);
    console.log(nums[0]);
    console.log(nums[2]);
    let popped = nums.pop();
    console.log(popped);
    console.log(nums.length);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: XPile 3-17/Programs/test_dynamic.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def main():
    nums = []
    nums.append(10)
    nums.append(20)
    nums.append(30)
    print(len(nums))
    print(nums[0])
    print(nums[2])
    popped = nums.pop()
    print(popped)
    print(len(nums))
    return 0


if __name__ == "__main__":
    main()
